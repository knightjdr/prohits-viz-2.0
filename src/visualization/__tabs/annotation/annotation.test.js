import React from 'react';
import { shallow } from 'enzyme';

import Annotation from './annotation';

const clearAnnotation = jest.fn();
const handleChange = jest.fn();
const handleClick = jest.fn();

describe('Annotation', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(
      <Annotation
        annotation="test"
        clearAnnotation={clearAnnotation}
        handleChange={handleChange}
        handleClick={handleClick}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle input change', () => {
    handleChange.mockClear();
    wrapper.find('Input').simulate('change');
    expect(handleChange).toHaveBeenCalled();
  });

  it('should set input value to prop', () => {
    const input = wrapper.find('Input');
    expect(input.props().value).toBe('test');
  });

  it('should handle annotation click', () => {
    handleClick.mockClear();
    wrapper.find('Button').first().simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });

  it('should handle clear click', () => {
    clearAnnotation.mockClear();
    wrapper.find('Button').at(1).simulate('click');
    expect(clearAnnotation).toHaveBeenCalled();
  });
});
