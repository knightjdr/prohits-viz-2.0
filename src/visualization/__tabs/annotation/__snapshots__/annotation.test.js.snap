// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation should match snapshot 1`] = `
<div
  className="visualization__annotation"
>
  <Input
    disabled={false}
    onChange={[Function]}
    placeholder="Annotation..."
    prefixCls="ant-input"
    type="text"
    value="test"
  />
  <Button
    className={null}
    disabled={false}
    onClick={[Function]}
    type="default"
  >
    Annotate
  </Button>
  <Button
    className={null}
    disabled={false}
    onClick={[Function]}
    type="warning"
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f12d",
            "M497.941 273.941c18.745-18.745 18.745-49.137 0-67.882l-160-160c-18.745-18.745-49.136-18.746-67.883 0l-256 256c-18.745 18.745-18.745 49.137 0 67.882l96 96A48.004 48.004 0 0 0 144 480h356c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12H355.883l142.058-142.059zm-302.627-62.627l137.373 137.373L265.373 416H150.628l-80-80 124.686-124.686z",
          ],
          "iconName": "eraser",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </Button>
</div>
`;
